lib/src/over_react_redux/over_react_flux.dart

```diff
@@ -0,0 +1,503 @@
+    mergeProps: mergeProps,
+    areOwnPropsEqual: areOwnPropsEqual,
+    areStatePropsEqual: areStatePropsEqual,
+    areMergedPropsEqual: areMergedPropsEqual,
+    context: context,
+    pure: pure,
+    forwardRef: forwardRef,
+    // Don't pass along the dispatcher as props by default
+    mapDispatchToProps: (_) => const {},
+    // This will always be false, since both arguments will also be the same store
+    areStatesEqual: (_, __) => false,
+  );
+}
+
+bool _shallowMapEquality(Map a, Map b) => const MapEquality().equals(a, b);
+
+/// A mixin that adds the fields necessary to make a Flux store compatible with the
+/// Influx architecture.
+///
+/// The mixin enforces the pattern necessary for an Influx architecture by adding
+/// the `influxReducer` and `triggerReduxUpdateFromFlux`, while also exposing the
+/// `state` field that is backed by a Redux state class. To make the store compatible
+/// with Redux, the Flux store using the [InfluxStoreMixin] should be wrapped
+/// with a [FluxToReduxAdapterStore].
+///
+/// __Example:__
+/// ```dart
+/// import 'package:w_flux/w_flux_server.dart' as flux;
+///
+/// class ExampleFluxStore extends flux.Store with InfluxStoreMixin<ReduxState> {
+///   FluxActions _actions
+///
+///   @override
+///   get reduxReducer => exampleReducer;
+///
+///   String get valueFromState => state.valueFromState;
+///
+///   ExampleFluxStore(this._actions) {
+///     state = ReduxState('default state');
+///
+///     triggerOnActionV2(_actions.fluxAction,
+///         () => this.influxReducer(ReduxAction('New Value')));
+///   }
+///
+/// }
+///
+/// class ReduxAction {
+///   String value;
+///
+///   ReduxAction(this.value);
+/// }
+///
+/// class ReduxState {
+///   String valueFromState;
+///
+///   ReduxState(this.valueFromState);
+/// }
+///
+/// ReduxState exampleReducer(ReduxState oldState, Object action) {
+///   if (action is ReduxAction) return ReduxState(action.value);
+///
+///   return oldState;
+/// }
+/// ```
+mixin InfluxStoreMixin<S> on flux.Store {
+  /// A traditional Redux reducer function that should return a new instance of
+  /// the corresponding state class.
+  ///
+  /// This is used in [influxReducer] to update the [state] field before triggering
+  /// an update.
+  redux.Reducer<S> get reduxReducer;
+
+  /// An instance of the Redux state model that the Flux store is migrating to.
+  S state;
+
+  /// A field to track if Flux has already tried to update Redux.
+  bool _isReduxInSync = false;
+
+  /// Checks to see if the Redux store needs to be updated and triggers an action
+  /// if so.
+  ///
+  /// This is only to be used from within the [FluxToReduxAdapterStore] to control
+  /// when Redux needs to receive an update to keep it in sync with Flux.
+  void triggerReduxUpdateFromFlux(Dispatcher dispatcher) {
+    // `state` can be null if `ConnectableFluxStore` is being used.
+    if (_isReduxInSync && state != null) {
+      _isReduxInSync = false;
+      return;
```

Comment:
💯 This exists because Flux naturally has an extra trigger due to `triggerOnActionV2`. I'll go through everything though:

This happens when a Flux action is triggered. In that case, the propagation would go 
1. triggerOnActionV2 
1. InfluxReducer
1. fluxStore direct mutation (updating the `state` field)
1. flux store trigger 
1. `FluxToReduxAdapterStore` event stream calls dispatch
1. Adapter store updates
1.  influxReducer short circuits

Then, `triggerOnActionV2` calls `trigger` again, and begins to repeat steps 5 through 7. But in this case, the original step 5 dispatch has gotten the store up to date so we can avoid triggering another one. At the same time, we need the trigger in the `influxReducer` for when `dispatch` is called directly on the adapter store

Reply:
Ah, that makes sense, thanks for explaining!! 

Could we capture this info somewhere in the code, possibly in the `_isReduxInSync` doc comment? I think you could just copy-paste what you wrote above?

Reply:
@greglittlefield-wf forgot to add this yesterday, but adding it now!